### PR TITLE
Fix build error under macOS.

### PR DESCRIPTION
**Pull request by Adversarr**
WARNING: Still will have runtime error, but able to build:
```
$ Binaries/Release/usdview_imgui_test
zsh: segmentation fault  Binaries/Release/usdview_imgui_test
```
and 
```
Coding Error: in Sync at line 57 of /Users/yangjerry/Repo/USTC_CG_2024_Dependencies/OpenUSD/pxr/imaging/hd/renderBuffer.cpp -- Bad dimensions for render buffer /_UsdImaging_HdStormRendererPlugin_0x1290db200/aov_depth
```

Screenshot 2024-03-22 at 20 38 20

### CODE REVIEW

File: Framework3D/configure.py
Comment (Jerry-Shen0527) [inline Framework3D/configure.py:1]: folder_list structure is different for Windows and Unix-Like systems so it should be a conditional choice

**Review by Jerry-Shen0527 — CHANGES_REQUESTED** [review]

**Comment by tiankaima** [comment]
Any luck?

I know this may have been out of date, here might be something useful to address the problem (tbh I never debugged any cg related projects)

## About OpenUSD on macOS(arm64):

Ref: https://github.com/PixarAnimationStudios/OpenUSD/issues/2559#issuecomment-1694676403

The tutorial recommend Roesseta and doesn't seems to fix the issue. Running 
```bash
python3 build_scripts/build_usd.py --ptex --openvdb  --embree  --prman  --openimageio  --opencolorio  --alembic  --hdf5  --draco --ignore-homebrew ~/USD
```
still won't work as some compile scripts aren't tuned for macOS(arm64).

## About ImGui & glfw

I've noticed this PR tries to adress the HiDPI issue as the following:

https://github.com/USTC-CG/USTC_CG_24/pull/14/files#diff-a94898fa46f45c85401daca2f13e5ef02e6e3213b58e54cdf33f0dcb058b09e3R149-R150

Setting to something like: (ref: https://github.com/ocornut/imgui/issues/5301#issuecomment-1122067363)
```c++
CGFloat scale = NSScreen.mainScreen.backingScaleFactor;
ImFontConfig cfg;
cfg.SizePixels = 13.0f * scale;
io.Fonts->AddFontDefault(&cfg);
io.FontGlobalScale = 1.0f / scale;
```

might offer a better solution

> Related discussions:
> - https://github.com/ocornut/imgui/issues/5081
> - https://github.com/ocornut/imgui/issues/5301

Anyways I'm dropping macOS and using Windows for this project, but do catch me up if there's any progress.

**Comment by Adversarr** [comment]
## OpenUSD Build

We have a tiny script for macOS[here](https://github.com/Jerry-Shen0527/USTC_CG_2024_Dependencies/blob/a8379c906f7adfb43235ee79a2c123cb4be50b89/build_mac.sh). I succeeded to build and run a basic demo, see line 230 for cmake arguments for configuring OpenUSD.

## HiDPI

I do not think HiDPI is a critical issue under macOS, and I prefer to keep the code clean and stable.

**Comment by tiankaima** [comment]
What about the texture issue? Here's what I got last week, exactly the issue. Have you figured it out?

090491c72a470a0e2ba2e9792added4a

**Comment by Adversarr** [comment]
I encountered this issue, but now it should have already been fixed (in my own fork https://github.com/Adversarr/USTC_CG_24/blob/main/Framework3D/source/GUI/usdview_engine.cpp)

image

**Pull request merged**